apps/extension/src/ui/hooks/useResolveEnsName.ts

```diff
@@ -0,0 +1,51 @@
+import { isEthereumAddress } from "@polkadot/util-crypto"
+import { isPotentialEns } from "@talismn/on-chain-id"
+import { useQuery } from "@tanstack/react-query"
+import { api } from "@ui/api"
+
+export const useResolveEnsName = (name?: string) => {
+  // check if name is something we can look up
+  const lookupName =
+    // don't look up undefined
+    name !== undefined &&
+    // don't look up ethereum addresses
+    !isEthereumAddress(name) &&
```

Comment:
Any chance we could use a more simple method like regex or string matching to identify an eth address here? `isEthereumAddress` from polkadot utils performs a cryptographic check on the value and is big and heavy. Ultimately I'd like to remove it from the frontend of the extension.